packages/gatsby/src/internal-plugins/functions/api-function-webpack-loader.ts

```diff
@@ -1,10 +1,11 @@
+import { slash } from "gatsby-core-utils"
 import type { LoaderDefinition } from "webpack"
 
 const APIFunctionLoader: LoaderDefinition = async function () {
   const params = new URLSearchParams(this.resourceQuery)
   const matchPath = params.get(`matchPath`)
 
-  const modulePath = this.resourcePath
+  const modulePath = slash(this.resourcePath)
```

Comment:
This just converts windows style file paths to unix one (so pretty much just `\` -> `/`), as windows style paths routinely has troubles with `\` escaping and unix style paths don't even have to deal with escaping problem